.github/workflows/artifacts.yml

```diff
@@ -0,0 +1,33 @@
+name: Upload Artifacts
+
+on:
+  push:
+    branches: [ master ]
+    paths:
+      - 'src/main'
```

Comment:
To my knowledge should this be `src/main/**` as GitHub would otherwise try to check if a file called `main` has been updated. The new pattern would accept any path after `src/main/`